File: Command/PHP/Resize.php
<?php
require_once('ICommand.php');
require_once('Image.php');

class Resize implements ICommand {
    private Image $target;
    private int $percentage;

    public function __construct(Image $target, int $percentage)
    {
        $this->target = $target;
        $this->percentage = $percentage;
    }

    public function execute() 
    {
        $this->target->resize($this->percentage);
    }
}

File: Command/Dart/Resize.dart
import 'ICommand.dart';
import 'Image.dart';

class Resize implements ICommand {
  Image? _target;
  int? _percentage;

  Resize(this._target, this._percentage);

  @override
  void execute() {
    var percentage = _percentage!;

    if (percentage != null) {
      _target?.resize(percentage);
    }    
  }
}
